### PR TITLE
Use Hive-style partitioning

### PR DESCRIPTION
**Pull request by jpdna**
Replaces #361 

Works with ADAM PR https://github.com/bigdatagenomics/adam/pull/1922

Reading partitioned files works, for example with command
```
mango-submit --master yarn --num-executors 10 --executor-cores 4 --executor-memory 20g --driver-memory 20g  -- /home/eecs/akmorrow/builds/hg19.2bit -genes http://www.biodalliance.org/datasets/ensGene.bb -reads hdfs://amp-bdg-master.amplab.net:8020/user/jpaschall/feb16_work/NA12889_S1.bam.partitioned.v4_withpartnum.adam -show_genotypes -parquetIsBinned
```

Note, this PR currently fails tests, but so does Mango Master for me at, 328b519
I get test failure
```
VizReadsSuite:
2018-02-25 16:29:33 WARN  NativeCodeLoader:62 - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
*** RUN ABORTED ***
  java.lang.NoClassDefFoundError: org/apache/http/ssl/SSLContexts
  at org.apache.http.impl.client.HttpClientBuilder.build(HttpClientBuilder.java:966)
  at org.scalatra.test.HttpComponentsClient$class.createClient(HttpComponentsClient.scala:99)
  at org.bdgenomics.mango.cli.VizReadsSuite.createClient(VizReadsSuite.scala:27)
  at org.scalatra.test.HttpComponentsClient$class.submit(HttpComponentsClient.scala:62)
  at org.bdgenomics.mango.cli.VizReadsSuite.submit(VizReadsSuite.scala:27)

```

### CODE REVIEW

**Comment by jpdna** [comment]
This worked fine with, ADAM with the PR https://github.com/bigdatagenomics/adam/pull/1948
at
f745bca

Note, you need to use partitioned parquet files generated from that version of the ADAM PR.
This works:
```
../mango/bin/mango-submit --master yarn --num-executors 10 --executor-cores 4 --executor-memory 20g --driver-memory 20g  -- ./hg19.2bit -genes http://www.biodalliance.org/datasets/ensGene.bb -reads hdfs://amp-bdg-master.amplab.net:8020/user/jpaschall/march9/NA12877_S1.partitioned.v2.adam  -show_genotypes 
```

**Comment by akmorrow13** [comment]
Jenkins, retest this please.

**Comment by akmorrow13** [comment]
@jpdna can you add unit tests?

File: mango-core/src/test/scala/org/bdgenomics/mango/models/VariantContextMaterializationSuite.scala [hunk 1]
```diff
@@ -67,6 +68,20 @@ class VariantContextMaterializationSuite extends MangoFunSuite {
 
   }
 
+  sparkTest("Can read Partitioned Parquet Genotypes") {
+
```
Comment (akmorrow13) [inline mango-core/src/test/scala/org/bdgenomics/mango/models/VariantContextMaterializationSuite.scala]: remove empty line

File: mango-core/src/test/scala/org/bdgenomics/mango/models/AlignmentRecordMaterializationSuite.scala [hunk 2]
```diff
@@ -51,9 +52,22 @@ class AlignmentRecordMaterializationSuite extends MangoFunSuite {
...
 
     val data = new AlignmentRecordMaterialization(sc, files, dict)
-
     val region = new ReferenceRegion("chrM", 0L, 900L)
-    val results = data.getJson(region).get(key).get
+    val results: Array[GAReadAlignment] = data.getJson(region).get(key).get
+  }
+
+  sparkTest("Read Partitioned Data") {
+
```
Comment (akmorrow13) [inline mango-core/src/test/scala/org/bdgenomics/mango/models/AlignmentRecordMaterializationSuite.scala]: remove empty line

**Comment by akmorrow13** [comment]
This looks great @jpdna ! Just some minor spacing comments, otherwise it looks good to go on my side.

**Comment by akmorrow13** [comment]
Replaced with https://github.com/bigdatagenomics/mango/pull/379

**Pull request closed without merging**